VS Code getting started tutorial (#7534)
microsoft/vscode-docs: docs/getstarted/getting-started.md

---
Order: 1
Area: getstarted
TOCTitle: Tutorial
ContentId: 72ad9b70-5227-4032-81d7-6aec00a1e8f8
PageTitle: Getting started with Visual Studio Code
DateApproved: 08/01/2024
MetaDescription: This tutorial gives you an overview of the key features of Visual Studio Code to help you get started quickly.
---
# Tutorial: Get started with Visual Studio Code

In this tutorial, you learn about the key features of Visual Studio Code to help you get started with coding quickly. You learn about the different components of the user interface and how to customize it to your liking. You then write some code and use the built-in code editing features, such as IntelliSense and Code Actions, and you learn about running and debugging your code. By installing a language extension, you add support for a different programming language.

<iframe src="https://www.youtube.com/embed/B-s71n0dHUk?autoplay=true" width="640" height="320" allowFullScreen="true" frameBorder="0" title="Getting Started with Visual Studio Code"></iframe>

## Prerequisites

- [Download Visual Studio Code](https://code.visualstudio.com/download)

> - [Set up Visual Studio Code on your computer](/docs/setup/setup-overview.md)

Reviewer: I was confused when I saw this link as I thought this current doc I'm on was the only doc I'd need to start with, and it made me wonder - how much reading does getting started with VS Code require?

When I check out that setup-overview doc, I see it's actually quite concise, and I'd even say I don't think users would have to read it to get started. To avoid users potentially feeling like getting started requires too much reading, I'd remove the setup doc from pre-reqs and maybe just list it in a specific paragraph(s) or troubleshooting section in case users need it for specific issues.

Author: Good point.

- [Install Node.js](https://nodejs.org/)

## Step 1: Open a folder in VS Code

> VS Code is a great lightweight editor where you can with individual files seamlessly, for example for quick edits or to view a file. However, by opening a folder, also known as a *workspace*, you unlock more features, such as folder-specific configuration settings, restoring the UI state for that folder, debug configurations, and more. Get more info about [workspaces](/docs/editor/workspaces.md).

Reviewer: Think this is missing a word like `work` `edit` or `code` 😄 

Personal nit: I might also consider avoiding "However" in the transition, as it gives almost a more negative or intimidating tone for my getting started.

```suggestion
VS Code is a great lightweight editor where you can work with individual files seamlessly, for example for quick edits or to view a file. Alternatively, by opening a folder, also known as a *workspace*, you unlock more features, such as folder-specific configuration settings, restoring the UI state for that folder, debug configurations, and more. Get more info about [workspaces](/docs/editor/workspaces.md).
```

Let's start by creating a folder and opening it in VS Code. You'll use this folder throughout the tutorial.

1. Create a new folder `vscode101` on your computer.

1. Open Visual Studio Code.

>     When you first open VS Code, you should see the **Welcome** page with different actions to get started. You can also explore some of the VS Code product walkthroughs.

Reviewer: I wonder if users will know what "product walkthroughs" are. I think there are questions even among more advanced users.

Author: Interesting. Because this is a guided experience, it might be better to remove the reference to product walkthroughs here and have a paragraph in our Tips & Tricks article that provides more details.

1. On the Welcome page, select **Open Folder...**, and then select the folder you created.

    The VS Code window reloads and you should see the folder name at the top of the **Explorer** view. You'll use the Explorer view to view and manage the files and folders in your workspace.

    ![Screenshot that shows VS Code after opening a folder, highlighting the Explorer view.](images/getting-started/open-folder.png)

    Alternatively, you can select **Open Folder...** from the **File** menu or press `kb(workbench.action.files.openFolder)` to open a folder.

> > **Tip**: When you use the command line, pass the folder name as a parameter to directly launch VS Code for that folder. For example, enter `code .` to open the current folder in VS Code.

Reviewer: nit: "When" sounds like users will be using the CLI at some point, but this wouldn't be a requirement right?

```suggestion
> **Tip**: If you use the command line to work with VS Code, pass the folder name as a parameter to directly launch VS Code for that folder. For example, enter `code .` to open the current folder in VS Code.
```

## Step 2: Explore the user interface

Now that you have a folder open in VS Code, let's take a quick tour of the user interface.

### Switch between views with the Activity Bar

The **Activity Bar** is located on the side of the window and gives you quick access to different views, such as the **Explorer**, **Search**, **Source Control**, and **Run and Debug** views.

1. Use the Activity Bar to explore the different views.

    As you hover over the Activity Bar, you can see the name of each view and the keyboard shortcut to open each view. You'll find that many of the features in VS Code have keyboard shortcuts already assigned to them, and you can also customize these shortcuts to your liking.

>     ![Screenshot that highlights the Activity Bar.](images/getting-started/activity-bar.png)

Reviewer: Love this screenshot. I think it's great you call out the hover info; I almost missed it in the screenshot, so I'd consider putting a red box around the hover info too.

1. Press `kb(workbench.action.quickOpen)` to use the Command Palette to navigate to a file in your workspace. Start typing to filter the list.

    The **Quick Open** feature enables you to quickly open a file in your workspace, or search for symbols in files. Get more info the [Command Palette](/docs/getstarted/userinterface.md#command-palette) in the VS Code documentation.

    ![Screenshot that shows the Quick Open feature in the Command Palette.](images/getting-started/quick-open.png)

### View status information with the Status Bar

The **Status Bar** is located at the bottom of the window and shows information about the file you are editing, and the workspace you have open. For example, if the folder you opened is a Git repository, the Status Bar shows the Git status and current branch.

1. Open a file by selecting it in the Explorer view.

    The Status Bar shows the language mode, indentation, and the line ending of the current editor.

    ![Screenshot that shows the Status Bar at the bottom of the window.](images/getting-started/status-bar.png)

> 1. Select the indentation field in the Status Bar to modify the indentation of the current file. For example, to use tabs instead of spaces.

Reviewer: Could be helpful to provide an example:

```suggestion
1. Select the indentation field (shown as `Spaces: 4` in the above image) in the Status Bar to modify the indentation of the current file. For example, to use tabs instead of spaces.
```

>     For a Git repository, you can select the branch name in the Status Bar and create a new branch.

Reviewer: Should we have an additional screenshot showing branch name in status bar?

Author: In this tutorial, users don't have source control added to their folder, so it will be confusing to highlight something that's not there.

## Step 3: Customize the user interface

> Each developer has their own preferences for how their coding environment should look like. VS Code enables you to change the layout, colors, keyboard shortcuts, and nearly every other aspect of the editor through various settings.

Reviewer: ```suggestion
Each developer has their own preferences for how their coding environment should look. VS Code enables you to change the layout, colors, keyboard shortcuts, and nearly every other aspect of the editor through various settings.
```

Let's start by using **Color Themes** to customize the colors in VS Code. A Color Theme affects both the VS Code user interface elements and the editor highlighting colors.

To select a different Color Theme:

> 1. Select the **Manage** button (gear icon) in the Activity Bar, and then select **Themes** > **Color Theme** to display the Color Theme picker.

Reviewer: Screenshot might be helpful (I think this is our first time mentioning the Gear in this doc?)

Author: It's is. Don't want to overload with too many screenshots. Maybe I can highlight it in the Activity Bar section? Good call-out.

1. Use the `kbstyle(Up)` and `kbstyle(Down)` keys to navigate through the list and preview the colors of the theme.

    VS Code comes with a list of built-in themes that you can choose from. As you move through the list, the active theme is previewed in VS Code.

    ![Screen capture that shows how to switch between different color themes by using the Command Palette.](images/themes/themes_hero.gif)

    > **Tip**: You can also select Color Themes from the VS Code Marketplace directly by selecting **Browse Additional Color Themes...** from the Color Theme picker.

Beyond changing the color theme of the user interface, you can customize almost every part of VS Code by using settings. Maybe you prefer a different font for the editor, change word wrapping, or enable [floating editor windows](/docs/editor/custom-layout.md#floating-editor-windows).

Let's use the **Settings Editor** to modify the tab size in the editor:

1. Select **File** > **Preferences** > **Settings** (or press `kb(workbench.action.openSettings)`) to open the Settings Editor.

    ![Screenshot that shows the Settings Editor.](images/getting-started/settings-editor.png)

> 1. Modify the value of the **Editor: Tab Size** setting, and notice how the change is applied directly to your editor.

Reviewer: How/where/when would I notice the change is applied to my editor?

## Step 4: Write some code

VS Code is first and foremost a code editor, so let's start by writing some code! VS Code has built-in support for JavaScript, TypeScript, HTML, CSS, and more. In this tutorial, you create a sample JavaScript file and use some of the code editing features that VS Code offers.

VS Code supports many programming languages and in the next step, you'll install a language extension to add support for a different language.

1. In the Explorer view, create a new file `app.js`, and start typing the following JavaScript code:

    ```javascript
    function sayHello(name) {
        console.log('Hello, ' + name);
    }

    sayHello('VS Code');
    ```

    As you type, you should see suggestions popping up that help you complete your code (*IntelliSense*). You can use the `kbstyle(Up)` and `kbstyle(Down)` keys to navigate the suggestions, and `kbstyle(Tab)` to insert the selected suggestion.

    Notice also the formatting of the code (*syntax highlighting*), to help you distinguish between different parts of the code.

    ![Screen capture that shows IntelliSense in action for a JavaScript file.](images/getting-started/javascript-intellisense.gif)

> 1. Put the cursor on the `Hello,` string, select the lightbulb icon, and then select **Convert to template string** to use a Code Action to simplify the string concatenation.

Reviewer: As a newer user, do I know what template string or simplified string concatenation mean?

Author: I'll add a note to clarify.

>     **Code Actions** are suggestions that help you apply quick fixes to your code. You can also use the `kb(editor.action.triggerSuggest)` keyboard shortcut to open the lightbulb menu.

Reviewer: I expected this explanation of Code Actions to go first (before I actually use one in the previous paragraph).

Author: Hmm, that makes total sense! Need to reshuffle some content.

    ![Screenshot that shows the lightbulb Code Action to convert a string concatenation to a template string.](images/getting-started/code-action-template-string.png)

    You can further experiment with IntelliSense in the different files in your workspace. For example, in the `index.html` file, start typing `<` to see suggestions for HTML tags.

    > **Tip**: Toggle the **File** > **Auto Save** menu item to automatically save files whenever you make changes.

> > **Tip**: You can further enhance your coding with artificial intelligence (AI) and [GitHub Copilot](/docs/copilot/overview.md). For example, get suggestions for lines of code or entire functions, perform code refactorings, generate code-related artifacts like tests, or produce code documentation, all from within the editor.

Reviewer: Maybe this shouldn't be formatted as a tip note since: 1) there's another tip note directly above it, 2) we'd consider Copilot important enough to be top-level info.

Reviewer: I wonder if Copilot should be its own top-level step at the very end? Considering how valuable / important we consider it. We could then also highlight ways it helps you get started with or configure your editor (i.e. `@vscode`).

Author: On it!

## Step 6: Run and debug your code

One of the key features in VS Code is its great support for running and debugging code. VS Code has built-in support for running and debugging Node.js applications, and there are extensions available for many other programming languages, such as Python, Java, C#, and more.

> **Note**: Make sure you have the necessary runtime and VS Code extensions installed on your computer. For example, to debug a Python program, you need to have Python installed on your computer.

In this tutorial, you debug the `app.js` file to explore the built-in Node.js debugger. Most debugging concepts and features are applicable to other debuggers as well.

Let's debug the `app.js` program by using the built-in Node.js debugger.

1. In the `app.js` file, put the cursor on the `console.log` line and press `kbstyle(F9)` to set a breakpoint.

    A red dot appears in the left margin of the editor, indicating that a breakpoint is set. With a breakpoint, you can pause the execution of your program at a specific line of code.

    ![Screenshot that shows a breakpoint in the editor.](images/getting-started/javascript-set-breakpoint.png)

> 1. Press `kbstyle(F5)` to start a debugging session, and then select the **Node.js** debugger from the list.

Reviewer: Should we show the list?

Author: Actually, when the `app.js` file is open, `F5` will automatically pick the Node.js debugger.

    Notice that the program starts and that the execution stops at the breakpoint you set.

    ![Screenshot that shows the program stopped at a breakpoint in the editor, highlighting the Variables view to inspect variables.](images/getting-started/vscode-debugging.png)

    > **Tip**: Inspect the value of the `name` variable by hovering over it in the editor while the execution is paused. You can view the value of variables at any time in the **Variables** view in the **Run and Debug** view.

1. Press the **Continue** button in the Debug toolbar or press `kbstyle(F5)` to continue the execution.

>     ![Screenshot that shows the Debug toolbar with the Continue button highlighted.](images/getting-started/debug-toolbar.png)

Reviewer: Image is broken?

> There are many more debugging features in VS Code, such as watch variables, conditional breakpoints, launch configurations, etcetera. Dive into the details of [debugging in VS Code](/docs/editor/debugging.md).

Reviewer: ```suggestion
There are many more debugging features in VS Code, such as watch variables, conditional breakpoints, and launch configurations. Dive into the details of [debugging in VS Code](/docs/editor/debugging.md).
```
